spec/requests/oauth/token_spec.rb

```diff
@@ -103,6 +105,53 @@
         end
       end
     end
+
+    context "with grant_type 'refresh_token'" do
```

Comment:
TODO: Add spec for reduce scope of an access token, and asserting the refresh token can be used to request back the full set of scopes afterwards (I suspect that Doorkeeper will actually be broken here)

So: 
1. 1st access token returned with `read write offline_access`
2. 1st refresh token used with `read`
3. 2nd access token should have `read` scope, but the 2nd refresh token should still have `read write offline_access` scope
4. 2nd refresh token is used with `read write offline_access` (expanding the scope back to original)
5. 3rd access token should have `read write offline_access` scope, since it was originally authorized by the access grant.

Reply:
Doorkeeper is broken here: https://github.com/doorkeeper-gem/doorkeeper/issues/1771